packages/core/src/integration.ts

```diff
@@ -40,24 +40,36 @@ function filterDuplicates(integrations: Integration[]): Integration[] {
 }
 
 /** Gets integrations to install */
-export function getIntegrationsToSetup(options: Pick<Options, 'defaultIntegrations' | 'integrations'>): Integration[] {
-  const defaultIntegrations = options.defaultIntegrations || [];
+export function getIntegrationsToSetup(
+  options: Pick<Options, 'defaultIntegrations' | 'integrations'>,
+  defaultIntegrations: Integration[] = [],
+): Integration[] {
   const userIntegrations = options.integrations;
 
+  // If `defaultIntegrations: false` is defined, we disable all default integrations
+  // TODO(v10): If an array is passed, we use this - this is deprecated and will eventually be removed
+  // Else, we use the default integrations that are directly passed to this function as second argument
+  const defaultIntegrationsToUse =
+    options.defaultIntegrations === false
+      ? []
+      : Array.isArray(options.defaultIntegrations)
+        ? options.defaultIntegrations
+        : defaultIntegrations;
```

Comment:
l: If we can't get my suggestion with `deprecated` to work, wdyt about adding a log here that passing an array will be removed in v10? Again, no super strong feelings but I'm fairly sure that people (mis-)use this option quite a lot which is why I'm thinking it's worth thinking about this.